pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -282,20 +282,33 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("A volume match does not exist for persistent claim: %s", claim.Name)
 		}
 
+		// create a reference to the claim and assign it to the volume being bound.
+		// the volume is a pointer and assigning the reference fixes a race condition where another
+		// claim might match this volume but before the claimRef is persistent in the next case statement
+		claimRef, err := api.GetReference(claim)
+		if err != nil {
+			return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
+		}
+
 		// make a binding reference to the claim.
-		// triggers update of the claim in this controller, which builds claim status
+		volume.Spec.ClaimRef = claimRef
```

Comment:
Do we?  Without the ref on the pointer, the race condition remains.  Before the watch gets the volume update, another claim could find that volume and bind to it.  Setting the value on the pointer prevents that, as it represents the current state.  

The store indexing these volumes does not index by this field.  If it needs to in the future, I believe I can update the store with the object myself so that it indexes correctly?